app/scripts/lib/middleware/pending.js

```diff
@@ -2,6 +2,8 @@ import { createAsyncMiddleware } from 'json-rpc-engine';
 import { formatTxMetaForRpcResult } from '../util';
 
 export function createPendingNonceMiddleware({ getPendingNonce }) {
+  // this would not be using the provider for the current domain...
```

Comment:
This should be okay as the network would be switched to the current domain when this method is called?

Reply:
It can also be called from dapps via `eth_getTransactionCount`. But in that case as well, it should be processed using the provider for the dapp selected network, so I think we're good there too.

Reply:
it ends up getting (this one)[ 
https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/metamask-controller.js#L5211] for `getPendingNonce` - which is not aware of domain.

Reply:
this is kind of getting into cross network accounts though... so yeah not sure how big a problem it really is, thats why I put this comment as a reminder to look into it